cgi-bin/DW/Controller/Manage/Circle/Invite.pm

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/perl
+#
+# DW::Controller::Manage::Circle::Invite
+#
+# /manage/circle/invite
+#
+# Authors:
+#      Cocoa <momijizukamori@gmail.com>
+#
+# Copyright (c) 2023 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself.  For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::Controller::Manage::Circle::Invite;
+
+use strict;
+
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+use DW::FormErrors;
+
+DW::Routing->register_string( "/manage/circle/invite", \&invite_handler, app => 1 );
+
+sub invite_handler {
+    my ( $ok, $rv ) = controller( authas => 1, form_auth => 1 );
+    return $rv unless $ok;
+    my $r    = DW::Request->get;
+    my $POST = $r->post_args;
+
+    my $remote = $rv->{remote};
+    my $u      = $rv->{u};
+
+    my @invitecodes;
+    my $code;
+    my $email_checkbox;
+    my $body = '';
+    my $create_link;
+
+    if ($LJ::USE_ACCT_CODES) {
+        @invitecodes = DW::InviteCodes->by_owner_unused( userid => $u->id );
+
+        if ( $u->is_identity ) {
+            return error_ml( '.error.openid', { sitename => $LJ::SITENAMESHORT } );
+        }
+
+        unless (@invitecodes) {
+            $body = LJ::Lang::ml('/manage/circle/invite.tt.msg.noinvitecodes');
+            $body .= " "
+                . LJ::Lang::ml( '/manage/circle/invite.tt.msg.noinvitecodes.requestmore',
+                { aopts => "href='$LJ::SITEROOT/invite'" } )
+                if DW::BusinessRules::InviteCodeRequests::can_request( user => $u );
+            return DW::Template->render_template( 'error.tt', { message => $body } );
+        }
+
+        $code = $POST->{code} || $invitecodes[0]->code;
+        $create_link .= "&code=" . $code;
+
+        # sort so that those which have been sent are last on the list
+        @invitecodes = sort { ( $a->timesent || 0 ) <=> ( $b->timesent || 0 ) } @invitecodes;
+    }
+
+    my $code_sent;
+    my $errors = DW::FormErrors->new;
+    if ( $r->did_post ) {
+        my $email = $POST->{'email'};
+        if ($email) {
+            my @errs;
+            LJ::check_email( $email, \@errs, $POST, \$email_checkbox );
+            $errors->add( "email", @errs ) if @errs;
+
+            if ( $LJ::USER_EMAIL && $email =~ /$LJ::USER_DOMAIN$/ ) {
+                $errors->add( "email", '.error.useralreadyhasaccount' );
+            }
+
+            unless ($LJ::USE_ACCT_CODES) {
```

Comment:
Hmm, another thing I just noticed. This block is nested within a `if ($LJ::USE_ACCT_CODES)` block, so it will never execute. I would suggest taking a top level view of your conditionals and making sure they're properly aligned.

Reply:
Oh okay, never mind then! My bad, I was just reviewing it in the browser window and didn't see where the cutoff was.